proto/conjunction.proto

```diff
@@ -0,0 +1,189 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+syntax = "proto3";
+
+import "proto/answer.proto";
+import "proto/concept.proto";
+
+package typedb.protocol;
+
+message ConjunctionStructure {
+  repeated StructureConstraint constraints = 1;
+  message Variable {
+    uint32 id = 1;
+  }
+
+  message StructureVertex {
+    oneof vertex {
+      Variable variable = 1;
+      Type label = 2;
+      Value value = 3;
+      NamedRole named_role = 4;
+      // Error condition. Should be unreachable.
+      string unresolved = 5;
+    }
```

Comment:
Much better handling of NamedRole.
Unresolved should currently never be reachable, but I've left it there for now.